core/src/main/java/nu/yona/server/subscriptions/service/UserAddService.java

```diff
@@ -262,13 +262,21 @@ UserDto addUserCreatedOnBuddyRequest(UserDto buddyUserResource)
 	}
 
 	@Transactional(dontRollbackOn = { MobileNumberConfirmationException.class, UserOverwriteConfirmationException.class })
-	void assertValidConfirmationCode(User userEntity, ConfirmationCode confirmationCode, String userProvidedConfirmationCode,
+	void assertValidConfirmationCode(User userEntity, Optional<ConfirmationCode> confirmationCode, String userProvidedConfirmationCode,
 			Supplier<YonaException> noConfirmationCodeExceptionSupplier,
 			IntFunction<YonaException> invalidConfirmationCodeExceptionSupplier,
 			Supplier<YonaException> tooManyAttemptsExceptionSupplier)
 	{
-		Require.isNonNull(confirmationCode, noConfirmationCodeExceptionSupplier);
+		assertValidConfirmationCode(userEntity, confirmationCode.orElseThrow(noConfirmationCodeExceptionSupplier),
+				userProvidedConfirmationCode, noConfirmationCodeExceptionSupplier, invalidConfirmationCodeExceptionSupplier,
+				tooManyAttemptsExceptionSupplier);
+	}
 
+	private void assertValidConfirmationCode(User userEntity, ConfirmationCode confirmationCode, String userProvidedConfirmationCode,
+			Supplier<YonaException> noConfirmationCodeExceptionSupplier,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "noConfirmationCodeExceptionSupplier". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS1172)